samcli/commands/check/resources/graph.py

```diff
@@ -0,0 +1,53 @@
+"""
+Class for graph. All data is stored in the graph directly, or within nodes that are stored in the graph
+"""
+from typing import List, Optional, Union
+
+from samcli.commands.check.resources.lambda_function_pricing import LambdaFunctionPricing
+from samcli.commands.check.resources.lambda_function import LambdaFunction
+from samcli.commands.check.resources.warning import CheckWarning
+
+
+class CheckGraph:
+    """
+    A graph object stores four main item types.
+
+    entry_points: All possible entry points to the application based on the template.
+    resources_to_analyze: All resources that need to be directly checked for bottle
+      neck issues (just lambda functions for now).
+    green/yellow/red/red_burst_warnings: The four different warning types that
+      the user can be presented
+    lambda_function_pricing_info: Contains the user entered data for the lambda
+      pricing questions
+    """
+
+    entry_points: List[Union[LambdaFunction]]
+    resources_to_analyze: List[Union[LambdaFunction]]
+    green_warnings: List[CheckWarning]
+    yellow_warnings: List[CheckWarning]
+    red_warnings: List[CheckWarning]
+    red_burst_warnings: List[CheckWarning]
+    lambda_function_pricing_info: Optional[LambdaFunctionPricing]
```

Comment:
As we discussed yesterday, these resource specific implementations shouldn't be leaked outside of its own class. We can think about moving them into interface definition or abstract classes. Ex;

```
unique_pricing_info: List[UniquePricingInfo]
```

```
class UniquePricingInfo(ABC):
    """Any common methods that we need to call can be defined in this class"""

class LambdaFunctionPricing(UniquePricingInfo):
    """
    This class will extend from UniquePricingInfo, and can implement the common methods 
    which will have Lambda specific business logic here
    """

```

Reply:
I agree. I will create a super class for all pricing objects. Instead of `unique_pricing_info` being a `List`, I think using a `dict` would keep things more organized. `unique_pricing_info = {"LambdaFunction": LambdaFunctionPricing}`.

Until pricing is implemented for more resources, the super class won't be used, as there aren't any common methods since lambda functions are the only resources with pricing implemented. The framework will still be there for other resources once they get pricing implemented for them.